File: crates/code-maps/tests/fixtures/go/sample.go
package sample

import "fmt"

type Project struct {
	Name string
}

func LoadProject(path string) (*Project, error) {
	return &Project{Name: path}, nil
}

func helper(count int) int {
	return count + 1
}

func (p *Project) NameLen() int {
	return len(p.Name)
}

func printProject(p Project) {
	fmt.Println(p.Name)
}

File: crates/code-maps/tests/fixtures/rust/sample.rs
use std::path::Path;

/// Loads a project from disk.
pub async fn load_project(path: &Path) -> Result<Project, Error> {
    Project::open(path).await
}

fn helper(count: usize) -> usize {
    count + 1
}

/// A project record.
pub struct Project {
    pub name: String,
}

pub type ProjectId = String;

const DEFAULT_LIMIT: usize = 25;
